### PR TITLE
Implement GraphData/GraphProfiler into DataProfiler Pipeline

### PR DESCRIPTION
**Pull request by MisterPNP**
- GraphData checked and identified in Data class
- GraphProfiler incorporated in Profiler input type checks
- Fix encoding issue in GraphData for utf-16+, used to default to utf-8 for all files
- Clearer readability for tests files

### CODE REVIEW

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 1]
```diff
@@ -171,7 +180,7 @@ def test_data_loader_nodes(self):
             options = dict()
             if not GraphData.is_match(input_file["path"], options):
                 return
-            data = GraphData(input_file["path"], options)
+            data = GraphData(input_file["path"], None, options)
```
Comment (taylorfturner) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:183]: might notate this as `data=None` just so folks know when they are reading this what that param is

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 2]
```diff
@@ -184,14 +193,22 @@ def test_data_loader_edges(self):
             all_edges_present = True
             if not GraphData.is_match(input_file["path"], options):
                 return
-            data = GraphData(input_file["path"], options)
+            data = GraphData(input_file["path"], None, options)
```
Comment (taylorfturner) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:196]: might notate this as data=None just so folks know when they are reading this what that param is

File: dataprofiler/data_readers/data.py [hunk 3]
```diff
@@ -17,6 +18,7 @@ class Data(object):
 
     data_classes = [
         dict(data_class=JSONData, kwargs=dict()),
+        dict(data_class=GraphData, kwargs=dict()),
         dict(data_class=CSVData, kwargs=dict()),
         dict(data_class=ParquetData, kwargs=dict()),
         dict(data_class=AVROData, kwargs=dict()),
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply dataprofiler/data_readers/data.py:24]: should be ordered this way actually to check Graph prior to CSV and other delimited datasets

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/tests/data_readers/test_csv_graph_data.py (hunk 2 above)
Comment (taylorfturner) [reply dataprofiler/tests/data_readers/test_csv_graph_data.py:196]: `input_file_path=input_file["path"]), data=None, options`

File: dataprofiler/tests/data_readers/test_csv_graph_data.py (hunk 1 above)
Comment (taylorfturner) [reply dataprofiler/tests/data_readers/test_csv_graph_data.py:183]: `input_file_path=input_file["path"]), data=None, options`

**Review by taylorfturner — APPROVED** [review]

**Review by JGSweets — COMMENTED** [review]
Initial thoughts.

File: dataprofiler/tests/profilers/test_graph_profiler.py [hunk 4]
```diff
@@ -110,7 +109,7 @@ def test_report(self):
         self.assertDictEqual(self.expected_profile, graph_profile.report())
 
     def test_graph_data_object(self):
-        data = GraphData(input_file_path=None, options=None, data=self.graph)
+        data = GraphData(input_file_path=None, data=self.graph, options=None)
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply dataprofiler/tests/profilers/test_graph_profiler.py:112]: agree

File: dataprofiler/tests/data_readers/test_csv_graph_data.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (taylorfturner) [reply dataprofiler/tests/data_readers/test_csv_graph_data.py]: resolved

File: dataprofiler/tests/data_readers/test_csv_graph_data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (taylorfturner) [reply dataprofiler/tests/data_readers/test_csv_graph_data.py]: resolved

File: dataprofiler/profilers/profile_builder.py [hunk 5]
```diff
@@ -11,9 +11,12 @@
 from collections import OrderedDict, defaultdict
 from datetime import datetime
 
+import networkx as nx
 import numpy as np
 import pandas as pd
 
+from dataprofiler.profilers.graph_profiler import GraphProfile
+
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply dataprofiler/profilers/profile_builder.py]: agree

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 6]
```diff
@@ -90,6 +91,14 @@ def setUpClass(cls):
 
         cls.file_or_buf_list = cls.input_file_names_pos + cls.buffer_list
 
+        cls.identify_file = [
+            dict(
+                path=os.path.join(test_dir, "csv/graph_data_csv_identify.csv"),
+                options={"header": 0, "delimiter": ","},
+                encoding="utf-8",
+            )
+        ]
```
[the remark this answers is not in the document]
Comment (JGSweets) [reply dataprofiler/tests/data_readers/test_csv_graph_data.py:100]: Any reason why this is separated from the above or can we just use the list above to do the test?

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 7]
```diff
@@ -184,14 +195,24 @@ def test_data_loader_edges(self):
...
             for edge in input_file["list_edges"]:
                 if edge not in data_edges and (edge[1], edge[0]) not in data_edges:
                     all_edges_present = False
             self.assertTrue(all_edges_present)
 
+    def test_factory_load(self):
+        """
+        Determine whether factory class Data identifies file correctly
+        """
+        for input_file in self.identify_file:
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:212]: can this just use the existing list?

**Review by tyfarnan — DISMISSED** [review]

File: dataprofiler/profilers/profile_builder.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (MisterPNP) [reply dataprofiler/profilers/profile_builder.py]: what do you mean? like this:
from dataprofiler.profilers.graph_profiler import (
     GraphProfile,
)
?

[the remark this answers is not in the document]
Comment (MisterPNP) [reply dataprofiler/profilers/profile_builder.py]: nvm fixed

**Review by taylorfturner — APPROVED** [review]

**Review by tyfarnan — APPROVED** [review]

**Pull request merged**